livekit_models.proto

```diff
@@ -391,8 +392,124 @@ enum ReconnectReason {
   RR_SWITCH_CANDIDATE = 4;
 }
 
+enum SubscriptionStatus {
+  Unsubscribed = 0;
+  Desired = 1;
+  Subscribed = 2;
+}
+
 enum SubscriptionError {
   SE_UNKNOWN = 0;
   SE_CODEC_UNSUPPORTED = 1;
   SE_TRACK_NOTFOUND = 2;
 }
+
+message ClientReport {
+
+  message RTCSenderStats {
+    // ice-candidate-pair
+    uint32 available_outgoing_bitrate = 1;
+    // local-candidate
+    string networkType = 3;
+    // outbound-rtp
+    uint64 packetsSent = 4;
+    uint32 packetsLost = 5;
+    uint32 nack_count = 6;    
+    uint32 retransmitted_packets_sent = 7;
+    uint64 timestamp = 8;
+    string stream_id = 9;
+  }
+
+  message RTCAudioSenderStats {
+   RTCSenderStats common_stats = 1;
+   // media-source
+   double total_audio_energy = 2;
+   double total_samples_duration = 3;
+  }
+
+  message RTCVideoSenderStats {
+   RTCSenderStats common_stats = 1;
+    // outbound-rtp
+    map<string, uint32> quality_limitation_durations = 2;
+    string quality_limitation_reason = 3;
+    uint32 quality_limitation_resolution_changes = 4;
+    string scalability_mode = 5;
+    uint32 frame_height = 6;
+    uint32 frame_width = 7;
+    uint32 frames_per_second = 8;
+    uint64 frames_sent = 9;
+    uint32 huge_frames_sent = 10;
+    string encoder_implementation = 11;
+    bool power_efficient_encoder = 12;
+  }
+
+  message RTCReceiverStats {
+    // inbound-rtp
+    double last_packet_received_timestamp = 1;
+    double jitter = 2;
+    int64 packets_lost = 3;
+  }
+
+  message RTCVideoReceiverStats {
+    RTCReceiverStats common_stats = 1;
+
+    // inbound-rtp
+    uint32 pli_count = 2;
+    uint32 sli_count = 3;
+    uint32 frames_decoded = 4;
+    uint32 frames_dropped = 5;
+    double total_decode_time = 6;
+    uint32 pause_count = 7;
+    double total_pauses_duration = 8;
+    uint32 freeze_count = 9;
+    double total_freezes_duration = 10;
+    uint32 fir_count = 11;
+    string decoder_implementation = 12;
+    bool power_efficient_decoder = 13;
+  }
+
+  message RTCAudioReceiverStats {
+    RTCReceiverStats common_stats = 1;
+    uint32 concealed_samples = 2;
+  }
+
+  message AudioPublicationReport {
+    string sid = 1;
+    RTCAudioSenderStats rtc_stats = 2;
+    int32 ready_state = 3;
+    string label = 4;
+    bool muted = 5;
+  }
+
+  message VideoPublicationReport {
+    string sid = 1;
+    RTCVideoSenderStats rtc_stats = 2;
+    int32 ready_state = 3;
```

Comment:
trying to understand these a bit:
* ready_state - I think this is just live or ended. do we expect to be ended?
* label - this seems a better fit as an event, since we do not expect this to change?
* muted - is this tracking MediaStreamTrack.muted property ? i.e. has data, or is it looking at our mute property? would be good to use a diff name here if it's former.

Reply:
1. we don't expect it to be `ended`, but it might be good debugging insight to know that for sure?
2. not sure why I wanted to include this in the first place, I guess we can drop it for v1 at least
3. it's looking at our `mute` property.